src/Components/Components/src/CascadingParameterState.cs

```diff
@@ -93,27 +93,16 @@ public static IReadOnlyList<CascadingParameterState> FindCascadingParameters(Com
         var candidateProps = ComponentProperties.GetCandidateBindableProperties(componentType);
         foreach (var prop in candidateProps)
         {
-            var attribute = prop.GetCustomAttribute<CascadingParameterAttribute>();
-            if (attribute != null)
-            {
-                result ??= new List<ReflectedCascadingParameterInfo>();
-
-                result.Add(new ReflectedCascadingParameterInfo(
-                    prop.Name,
-                    prop.PropertyType,
-                    attribute.Name));
-            }
-
-            var hostParameterAttribute = prop.GetCustomAttributes()
-                .OfType<IHostEnvironmentCascadingParameter>().SingleOrDefault();
-            if (hostParameterAttribute != null)
+            var cascadingParameterAttribute = prop.GetCustomAttributes()
+                .OfType<ICascadingParameterAttribute>().SingleOrDefault();
+            if (cascadingParameterAttribute != null)
```

Comment:
We need to think about what happens when you have [SupplyParameterFromQuery] and [SupplyParameterFromFrom] attribute. There are some cases that might be "incompatible", but there are other cases where we might want both attributes to apply. For example, if we do `[SupplyParameterFromPrerenderedState]`

Reply:
Hm, true. I wonder if it would be fine to modify this code slightly to add a `ReflectedCascadingParameterInfo` for each attribute on the parameter property. Maybe we could also add something like an `Order` property to `ICascadingParameterAttribute` so that, for example, if `[SupplyParameterFromQuery]` and `[SupplyParameterFromForm]` are both present, we prioritize supplying a value from a form when possible, but fall back to supplying a parameter from the query if form data is not available.

Is this something you think should be addressed by this PR?

Reply:
Great points to raise. It would definitely be consistent with other parts of the framework if we had an `Order` property.

I don't think you should feel required to implement that in this PR but if there's broad agreement about the direction, it would be good to file an issue summarizing the design then we can hopefully get on and implement that soon.

Reply:
BTW we likely end up with a fairly clear, straightforwards design if:

 * `CascadingParameterAttribute` is unsealed
 * ... and gains a new `Order` property
 * ... and we change it to `AllowMultiple = true`

Then we can cheaply have form/query subclasses in a different assembly, and the core doesn't need to know or care about anything but the base class.

Reply:
`CascadingParameterAttribute` I'd rather not open up the ability for third-parties to create their own definitions of Cascading values by extending/overriding CascadingParameterAttribute.

Reply:
> Then we can cheaply have form/query subclasses in a different assembly, and the core doesn't need to know or care about anything but the base class.

I like this, but wouldn't `CascadingModelBinder` and `CascadingQueryValueProvider` still need to know what attribute type was used so that, in the case of only one attribute being present, they don't supply values for each other (or for `CascadingValue`) by accident? Or would we be okay with, for example, a parameter annotated with `[CascadingValue]` receiving values from the query or a form request?

Reply:
I would expect `CascadingModelBinder` and `CascadingQueryValueProvider` to be in layers where they are able to know about the corresponding `CascadingParameterAttribute` subclass and hence can do the same type check that you already have.